crates/consensus/beacon/src/engine/sync.rs

```diff
@@ -304,6 +304,10 @@ where
 
         // drain an element of the block buffer if there are any
         if let Some(block) = self.range_buffered_blocks.pop() {
+            // peek ahead and pop duplicates
+            while self.range_buffered_blocks.peek() == Some(&block) {
+                _ = self.range_buffered_blocks.pop();
+            }
```

Comment:
I think it's sufficient to only compare the hashes here

Reply:
I'm not sure how to do that in a good way since `peek` returns an option.

A `while let` is going to require another inner conditional as far as I can tell and I didn't want to nest any deeper.

I'm happy to use any suggestions you guys have.

Reply:
ah okay,

since we want to find duplicates, we need to check whether the `peek` block's hash is the same as the block that we already popped (duplicate)

Reply:
```rust
// peek ahead and pop duplicates
while self
    .range_buffered_blocks
    .peek()
    .unwrap_or(&Reverse(OrderedSealedBlock(SealedBlock::default()))).0.0
    .hash()
    == block.0 .0.hash()
{
    _ = self.range_buffered_blocks.pop();
}
```
Is this what you have in mind?

Reply:
yeah, but we can make this a bit nicer by using peek_mut


while let Some(peek) peek_mut {
  // check hashes
  if hashes identical {
   peek.pop();
   continue
  }
   
  break
}

Reply:
ah, yeah peek_mut makes this nice

Reply:
Sorry, I'm not following - how do we pop on a `PeekMut`?

```rust
while let Some(peek) = self.range_buffered_blocks.peek_mut() {
    if peek.0 .0.hash() == block.0 .0.hash() {
        peek.pop();
    } else {
        break;
    }
}
```

no `pop` method on `peek`.

Reply:
`PeekMut::pop(peek)`

Reply:
Ahh thanks. Will commit.